src/debug/ee/functioninfo.cpp

```diff
@@ -1558,30 +1560,77 @@ DebuggerJitInfo *DebuggerMethodInfo::FindOrCreateInitAndAddJitInfo(MethodDesc* f
 
     _ASSERTE(fd != NULL);
 
-    // This will grab the latest EnC version.
-    TADDR addr = (TADDR) g_pEEInterface->GetFunctionAddress(fd);
+    // The debugger doesn't track Lightweight-codegen methods b/c they have no metadata.
+    if (fd->IsDynamicMethod())
+    {
+        return NULL;
+    }
 
-    if (addr == NULL)
+#ifdef FEATURE_CODE_VERSIONING
+    bool isStartAddrLatest = false;
+#endif
+    if (startAddr == NULL)
+    {
+        // This will grab the latest EnC version.
+        startAddr = (TADDR)g_pEEInterface->GetFunctionAddress(fd);
+        if (startAddr == NULL)
+        {
+            return NULL;
+        }
+#ifdef FEATURE_CODE_VERSIONING
+        isStartAddrLatest = true;
+#endif
+    }
+#ifndef FEATURE_CODE_VERSIONING
+    else if (startAddr != (TADDR)g_pEEInterface->GetFunctionAddress(fd))
+    {
+        // Only one code version is available and the requested start address does not match
         return NULL;
+    }
+#endif
 
     // Check the lsit to see if we've already populated an entry for this JitInfo.
     // If we didn't have a JitInfo before, lazily create it now.
     // We don't care if we were prejitted or not.
     //
     // We haven't got the lock yet so we'll repeat this lookup once
     // we've taken the lock.
-    DebuggerJitInfo * pResult = FindJitInfo(fd, addr);
+    DebuggerJitInfo * pResult = FindJitInfo(fd, startAddr);
     if (pResult != NULL)
     {
-        // Found!
         return pResult;
     }
 
+#ifdef FEATURE_CODE_VERSIONING
+    if (!isStartAddrLatest)
```

Comment:
I'm not sure what the purpose of this block is? foundStartAddr never gets used below and if there was a case where the code version manager didn't know about an address but it is valid jitted code with a DJI I don't think we'd want to return NULL?

Reply:
Ah now I understand your concern. I think we should expect callers to pass a reasonable IP, but in DEBUG builds you could add something like:

```
EECodeInfo codeInfo(startAddr);
if(codeInfo.GetMethodDesc() != fd)
{
    _ASSERTE(!"Address is not a valid IP for this method")
    return NULL;
}
```